modules/compiler/vecz/source/include/vectorization_context.h

```diff
@@ -150,14 +153,45 @@ class VectorizationContext {
   /// @return The masked version of the function
   llvm::Function *getOrCreateMaskedFunction(llvm::CallInst *CI);
 
+  struct MaskedAtomicRMW {
+    llvm::Type *PointerTy;
+    llvm::Type *ValTy;
+    llvm::AtomicRMWInst::BinOp BinOp;
+    llvm::Align Align;
+    bool IsVolatile = false;
+    llvm::SyncScope::ID SyncScope;
+    llvm::AtomicOrdering Ordering;
+    // Vectorization info
+    llvm::ElementCount VF;
+    bool IsVectorPredicated = false;
+  };
+
+  /// @brief Check if the given function is a masked version of an atomic RMW
+  /// operation.
+  ///
+  /// @param[in] F The function to check
+  /// @return true if the function is a masked atomic RMW, or false otherwise
+  std::optional<MaskedAtomicRMW> isMaskedAtomicRMWFunction(
```

Comment:
There is a mismatch between the documentation and the implementation here, the return type does not allow a `true`/`false` result.

Reply:
Thanks, updated!